### DISCUSSION TITLE
Sensor update frequency

### DISCUSSION THREAD

**Discussion by leonbezuidenhout** [kellerza/sunsynk#84] [General]
Hi KellerZA

Excellent integration, thank you for the work!

How often is the sensor values updated in Home Assistant, e.g. every second, 5 seconds? And can this update frequency be changed for specific sensors?

I'm using the integration to switch off "heavy" loads during loadshedding (e.g kettles, aircons) to prevent AC overload on the inverter. So the quicker the Automatic triggers on e.g. Grid Power (when grid goes to zero), the better chance it has to prevent a trip. Ideally I need it to switch off smart switches within 5 seconds, so that I can stay within the inverters 10s peak power allowance.

Regards

**Comment by kellerza** [comment]
By default, the step sensor should allow you to switch such load in less than 2 seconds. If there are no significant steps it will only update every 60 seconds.

You can change it, I just added a comment with more detail in #70

**Comment by leonbezuidenhout** [comment]
Thanks. I'll leave it at the default settings then - 2 seconds is good enough for my application.

**Reply by kellerza** [reply]
Better to test what you want to do, if you draw very little power form the grid and it fails, you won't see a change!

But if you look at `grid_connected_status` or even `grid_frequency:step:10` it should be fairly quick.

**Comment by kellerza** [comment]
You should be able to test the ones you are interested in by keeping you HASS app open and switching you kettle/load on/off
